### PR TITLE
feat(tsgen): tsgen export buildTypes function

### PR DESCRIPTION
**Pull request by jaronnie**

### CODE REVIEW

**Comment by kevwan** [comment]
Why to export it? Would you please describe it in details?

**Comment by jaronnie** [comment]
image
Because I want to generate typescript sdk by api file, and generate ts model by api type.

**Review by kesonan — APPROVED** [review]
LGTM

File: tools/goctl/api/tsgen/gencomponents.go [hunk 1]
```diff
@@ -62,3 +62,7 @@ func buildTypes(types []spec.Type) (string, error) {
 
 	return builder.String(), nil
 }
+
+func BuildTypes(types []spec.Type) (string, error) {
```
Comment (kevwan) [inline tools/goctl/api/tsgen/gencomponents.go:66]: Why not just change the original `buildTypes` to `BuildTypes`?

Comment (kevwan) [reply tools/goctl/api/tsgen/gencomponents.go:66]: @jaronnie Is there any concern on adding a new method instead of exporting the original one?

Comment (jaronnie) [reply tools/goctl/api/tsgen/gencomponents.go:66]: Minimal changes

Comment (kevwan) [reply tools/goctl/api/tsgen/gencomponents.go:66]: I recommend to remove the newly added one, and export the previous method.

**Pull request merged**